hack/netpol/pkg/main/main.go

```diff
@@ -55,7 +60,22 @@ func waitForPodInNamespace(k8s *Kubernetes, ns string, pod string) error {
 		if err != nil {
 			return errors.WithMessagef(err, "unable to get pod %s/%s", ns, pod)
 		}
+
 		if k8sPod != nil && k8sPod.Status.Phase == v1.PodRunning {
+			if k8sPod.Status.PodIP == "" {
+				return errors.WithMessagef(err, "unable to get IP of pod %s/%s", ns, pod)
+			} else {
+				log.Debugf("IP of pod %s/%s is: %s", ns, pod, k8sPod.Status.PodIP)
+				podIPs[ns+"/"+pod] = k8sPod.Status.PodIP
+			}
+			//Get podCIDR from node where the pod is running
+			podCIDR, err := k8s.GetPodCIDRFromNode(k8sPod.Spec.NodeName)
```

Comment:
@abhiraut @tnqn @jayunit100 how do you feel about adding a dependency to podCIDR in netpol? It seems that the long term plan may be to get rid of it (https://github.com/kubernetes/kubernetes/issues/57130).

Not saying I have a better plan for testing IP blocks...

Reply:
in a way, with this change, the effectiveness of the tests depends on which nodes the pods are running on, which is not great. maybe a better solution would be to add services into the framework. if we set the clusterIP field in the service spec, we can test IP blocks in a better fashion possibly?

Reply:
me neither.. this will also probably fail for CNIs that do not rely on host local IPAM for Pod, for example Calico, who do their own IPAM. In case of Antrea, we may also do our own IPAM in near future.. can you use an alternate where you generate the PodCidr based on the PodIP and use that for testing?

Reply:
Thanks @antoninbas and @abhiraut for comments.
Good to know about the side-effects of having podCIDR dependency. Removed this dependency by building own allowCIDR as per suggestion.
I did not think in the lines that test case should pertain to an usecase, where network policies with CIDRs are used. It makes sense as netpol is planned to get upstreamed. 
Service deployment framework with cluster IPs sounds interesting. Is the use case for clusterIP CIDRs is similar to multi-tenancy that can be achieved through T0/T1 routers in NSX?
As the service framework is not available in netpol and this probably needs a different truth table like CIDR-to-CIDR instead of pod-to-pod. Can we take this up as separate issue?